src/snowcli/cli/loggers.py

```diff
@@ -1,32 +1,113 @@
 import logging
+from logging.handlers import TimedRotatingFileHandler
+from pathlib import Path
 
 import typer
+from snowcli.config import (
+    get_logs_config,
+    is_default_logs_path,
+)
+from snowcli.exception import InvalidLogsConfiguration
 
 DEFAULT_LOG_FORMAT = "%(asctime)s %(levelname)s %(message)s"
 DEBUG_LOG_FORMAT = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
+FILE_LOG_FORMAT = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+_CONSOLE_OUTPUT = "console output"
+
+
+class LogsConfig:
+    def __init__(self, debug: bool) -> None:
+        config = get_logs_config()
+
+        self.path: Path = Path(config["path"])
+        self.save_logs: bool = config["save_logs"]
+        self.file_log_level: int = logging.getLevelName(config["level"].upper())
+        if debug:
+            self.file_log_level = logging.DEBUG
+
+        self._check_log_level(config)
+        if self.save_logs:
+            self._check_logs_directory_exists()
+
+    def _check_logs_directory_exists(self):
+        if not self.path.exists():
+            if is_default_logs_path(self.path):
+                self.path.mkdir(parents=True)
+            else:
+                raise InvalidLogsConfiguration(
+                    f"Directory '{self.path}' does not exist"
+                )
+
+    def _check_log_level(self, config):
+        possible_log_levels = [
+            logging.DEBUG,
+            logging.INFO,
+            logging.WARN,
+            logging.ERROR,
+            logging.CRITICAL,
+        ]
+        if self.file_log_level not in possible_log_levels:
+            raise InvalidLogsConfiguration(
+                f"Invalid 'level' value set in [logs] section: {config['level']}. "
+                f"'level' should be one of: {' / '.join(logging.getLevelName(lvl) for lvl in possible_log_levels)}"
+            )
+
+
+def remove_console_output_handler_from_logs() -> None:
+    logger = logging.getLogger("snowcli")
+    for handler in logger.handlers:
+        if handler.name == _CONSOLE_OUTPUT:
+            logger.removeHandler(handler)
+
+
+def add_console_output_handler_to_logs(
+    log_level: int, formatter: logging.Formatter
+) -> None:
+    logger = logging.getLogger("snowcli")
+    console = logging.StreamHandler()
+    console.set_name(_CONSOLE_OUTPUT)
+    console.setFormatter(formatter)
+    console.setLevel(log_level)
+    logger.addHandler(console)
```

Comment:
I'm wondering if we can utilise `dictConfig` to store more structured information about loggers
https://docs.python.org/3/library/logging.config.html#logging-config-dictschema 

One of the examples I know:
https://github.com/apache/airflow/blob/95a83102e8753c2f8caf5b0d5c847f4c7f254f67/airflow/config_templates/airflow_local_settings.py#L66-L140

Maybe it would be useful to define multiple configurations? What do you think?

Reply:
Nice one! Definitely increases readability - done